src/uffs/uffs_tree.c

```diff
@@ -588,8 +588,10 @@ static URET _BuildTreeStepOne(uffs_Device *dev)
 							"I/O error when reading mini header !"
 							"block %d page %d",
 							block, 0);
-				ret = U_FAIL;
-				break;
+				node->u.list.block = block;
+				uffs_BadBlockProcessNode(dev, node);
```

Comment:
Call `uffs_BadBlockProcessNode` only when node has been added to bad block pending list

Reply:
Fixed